model_zoo.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# @Organization  : insightface.ai
+# @Author        : Jia Guo
+# @Time          : 2021-05-04
+# @Function      : 
+
+import os
+import os.path as osp
+import glob
+import onnxruntime
+from .arcface_onnx import *
+from .scrfd import *
+
+#__all__ = ['get_model', 'get_model_list', 'get_arcface_onnx', 'get_scrfd']
+__all__ = ['get_model']
+
+
+class ModelRouter:
+    def __init__(self, onnx_file):
+        self.onnx_file = onnx_file
+
+    def get_model(self):
+        session = onnxruntime.InferenceSession(self.onnx_file, providers=['CUDAExecutionProvider', 'CPUExecutionProvider'])
+        input_cfg = session.get_inputs()[0]
+        input_shape = input_cfg.shape
+        outputs = session.get_outputs()
+        #print(input_shape)
+        if len(outputs)>=5:
+            return SCRFD(model_file=self.onnx_file, session=session)
+        elif input_shape[2]==112 and input_shape[3]==112:
+            return ArcFaceONNX(model_file=self.onnx_file, session=session)
+        else:
+            raise RuntimeError('error on model routing')
+
+def find_onnx_file(dir_path):
+    if not os.path.exists(dir_path):
+        return None
+    paths = glob.glob("%s/*.onnx" % dir_path)
+    if len(paths) == 0:
+        return None
+    paths = sorted(paths)
+    return paths[-1]
+
+def get_model(name, **kwargs):
```

Comment:
You added this file but don't use it, because the model_zoo function is imported from the from insightface module.
I'd rather not add this file to the repo at all.

Reply:
I think we don't need the SimSwapInference, just simply remove it from the import paths.

Reply:
Yes, model_zoo.py was added for the Colab Notebook, to make it easier to use.
Placing src folder in SimSwapInference folder is just a way to fix it, I tested it on Colab and it is working, you can make your own fixes of course if you don't like this way.
Check Discord 👍